### PR TITLE
add support for Users

### PR DESCRIPTION
**Pull request by leogermani**
This PR adds support for users functions :) 

Here is a non-exaustive list of functions supported:

* `wp_insert_user`
* `wp_update_user`
* `wp_delete_user`
* `get_userdata`
* `new WP_User( $id )` to fetch a user
* `user_can`
* `current_user_can`
* `set_current_user`
* `get_current_user_id`
* `wp_get_current_user`
* `get_user_meta`
* `update_user_meta`
* `add_user_meta`
* `delete_user_meta`

## Test instructions

* Look at the code (note: `Metadata.php` is just code moved around, disconsider)
* Look at `test-users.php` in the tests
* If you think of something else, try writing a couple of tests emulating a use case.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Needs another PR...

**Comment by szepeviktor** [comment]
Now what?
How to press that <kbd>Merge</kbd> button?

**Comment by leogermani** [comment]
> Now what?
> How to press that Merge button?

I'm working on real tests using this here -> https://github.com/Automattic/jetpack/pull/17832

Let me experiment more with this and I'll probably find issues, as I already did

File: src/ClearCacheGroup.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace WorDBless;
+
+trait ClearCacheGroup {
```
Comment (szepeviktor) [inline src/ClearCacheGroup.php:5]: Very nice trait!

File: README.md [hunk 2]
```diff
@@ -90,7 +90,32 @@ Manipulating (creating, updating, deleting) posts and attachments will work. Fet
 * `wp_get_attachment_image`
 * and almost anything related to the manipulation of one attachment
 
-Note: Fetching for posts using `WP_Query` will not work (yet)! To get a post use `get_post( $id )`.
+Note: Fetching posts using `WP_Query` will not work (yet)! To fetech a post, use `get_post( $id )`.
```
Comment (szepeviktor) [inline README.md:93]:
```suggestion
Note: Fetching posts using `WP_Query` will not work (yet)! To fetch a post, use `get_post( $id )`.
```

Another typo.

**Pull request merged**